### PR TITLE
file_create_dir() rewritten 

### PR DESCRIPTION
**Pull request by hujung-de**
file : <nanomq/nanolib/file.c>
function file_create_dir() completely rewritten ... old function renamed as file_create_dir_DEP()

### CODE REVIEW

**Comment by JaylinYu** [comment]
Please abide to our code format (clang-format)

File: nanolib/file.c [hunk 1]
```diff
@@ -147,8 +147,64 @@ file_delete(const char *fpath)
...
+  }
+	
+  debug_msg("DBG: %s(%s) : OK",__func__,pName);
+  free(path);
+  return (0);
+}	
+
 int
-file_create_dir(const char *fpath)
+file_create_dir_DEP(const char *fpath)
```
Comment (wanghaEMQ) [inline nanolib/file.c:207]: What does DEP mean?

File: nanolib/file.c [hunk 2]
```diff
@@ -147,8 +147,64 @@ file_delete(const char *fpath)
...
+    slash += strspn (slash, "/");
+    slash += strcspn(slash, "/");
+    if (*slash == '\0') {
+      done = 1;	    
+    }
+    *slash = '\0';
+
+    if (stat(path, &sb)) {
+      if (errno != ENOENT || (mkdir(path, 0777) && errno != EEXIST)) {
+	 debug_msg("ERR: %s(%s) : %s",__func__,path,strerror(errno));
```
Comment (wanghaEMQ) [inline nanolib/file.c:187]: If you are in debug mode and the log is on, the function name will be printed by debug_msg()

File: nanolib/file.c (hunk 1 above)
Comment (hujung-de) [reply nanolib/file.c:207]: good morning, i use DEP  as the abbreviation for "DEPricated" if i want to remain the previous function code

File: nanolib/file.c (hunk 2 above)
Comment (hujung-de) [reply nanolib/file.c:187]: ok ... will change

**Review by wanghaEMQ — CHANGES_REQUESTED** [review]
Thanks for your support to Nanomq, format your code and rename the function, and request again.

File: nanolib/file.c (hunk 1 above)
Comment (wanghaEMQ) [reply nanolib/file.c:207]: Maybe you can delete the previous function code ...

Comment (wanghaEMQ) [reply nanolib/file.c:207]: oh, by the way, don't forget to beautify your commits :)

**Comment by hujung-de** [comment]
old code of file_create_dir() removed, hopefully formatted in the right way & and removed the function name in the debug_msg() call

**Review by wanghaEMQ — APPROVED** [review]
Thanks for your commit :)

**Pull request merged**